### PR TITLE
RC_Channel: Correct RC PWM min to 800 and make RC PWM LIMITS variables

### PR DESCRIPTION
**Pull request by khancyr**
Rework from https://github.com/ArduPilot/ardupilot/pull/13436

Set the RC PWM limit correctly according to SRXL protocol that support 800 - 2200 bounds : https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RCProtocol/AP_RCProtocol_SRXL.cpp#L71
Comments are fine.

close https://github.com/ArduPilot/ardupilot/pull/13436

### CODE REVIEW

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: libraries/RC_Channel/RC_Channel.cpp [hunk 1]
```diff
@@ -1195,7 +1195,7 @@ void RC_Channel::init_aux()
 bool RC_Channel::read_3pos_switch(RC_Channel::AuxSwitchPos &ret) const
 {
     const uint16_t in = get_radio_in();
-    if (in <= 900 or in >= 2200) {
+    if (in <= RC_MIN_LIMIT_PWM or in >= RC_MAX_LIMIT_PWM) {
```
Comment (rmackay9) [inline libraries/RC_Channel/RC_Channel.cpp:1198]: my goodness, what is this "or" doing in here?  obviously an existing problem but we should get rid of these.

It looks like the felony was committed by @peterbarker with me as his accomplice! PR https://github.com/ArduPilot/ardupilot/pull/10199

**Comment by tridge** [comment]
discussed on eu dev call, decided to leave lower bounds that were at 900 at the same 900 value

**Review by rmackay9 — CHANGES_REQUESTED** [review]
I think we should leave the lower limit at 900 just to be safe.  There's no real advantage to having it accept 800 I think.

File: libraries/RC_Channel/RC_Channel.cpp (hunk 1 above)
Comment (khancyr) [reply libraries/RC_Channel/RC_Channel.cpp]: changed !

**Comment by khancyr** [comment]
changed done!

**Comment by Hwurzburg** [comment]
just a question: this in no way limits the lower PWM value we can drive an output using passthru, correct? we need to be able to drive outputs to 500us for servo stretching....this is only input range validation, correct?

**Comment by rmackay9** [comment]
@Hwurzburg,

That's right, this only affects the interpretation of the mode channel's input.  In short users will find that a very low mode channel input will not change the mode to whatever is in the FLTMODE1 parameter.  There's no change in behaviour though actually... this is now a non-functional change.

**Comment by tridge** [comment]
@rmackay9 needs your re-review

**Review by rmackay9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**